src/controllers/CategoryController.ts

```diff
@@ -9,22 +9,21 @@ export class CategoryController {
     @inject(CategoryService) private categoryService: CategoryService
   ) {}
 
-  public async getAllCategories(
-    req: Request,
-    res: Response
-  ): Promise<Category[]> {
+  public async getAllCategories(req: Request, res: Response) {
     try {
       const Categories = await this.categoryService.getAllCategories()
-
-      res.json(Categories)
-      return Categories
+      if (!Categories) {
+        return res.status(404).json({ error: 'could not find categories' })
+      }
+      return res.status(200).json(Categories)
     } catch (error: any) {
-      res.status(500).json({ error: error.message })
-      throw new Error(`Error retrieving categories`)
+      res
+        .status(500)
+        .json({ error: 'Internal server error, please try again later' })
     }
   }
 
-  public async getCategoryByID(req: Request, res: Response): Promise<void> {
+  public async getCategoryByID(req: Request, res: Response) {
     try {
       const categoryId = parseInt(req.params.id, 10)
```

Comment:
What does the number 10 refer to ? I think it may be better to have it in consts file and give it a name

Reply:
We don't need it anymore since we built validators for inputs